kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java

```diff
@@ -1536,15 +1579,29 @@ private Att addKoreAttributes(Production prod, SetMultimap<KLabel, Rule> functio
         String format = att.getOptional("format").orElse(Formatter.defaultFormat(prod.items().size()));
         int nt = 1;
         boolean hasFormat = true;
+        boolean printName = stream(prod.items()).noneMatch(pi -> pi instanceof NonTerminal && ((NonTerminal) pi).name().isEmpty());
+        boolean printEllipses = false;
+
         for (int i = 0; i < prod.items().size(); i++) {
           if (prod.items().apply(i) instanceof NonTerminal) {
-            format = format.replaceAll("%" + (i+1) + "(?![0-9])", "%" + (nt++));
+            String replacement;
+            if (printName && isPrefixProduction(prod.items())) {
```

Comment:
```suggestion
            if (printName && prod.isPrefixProduction()) {
```
The original code is accessible from here. No need to duplicate it.